src/main/java/ladder/domain/Line.java

```diff
@@ -0,0 +1,39 @@
+package ladder.domain;
+
+import java.util.stream.IntStream;
+import ladder.dto.LineResult;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.function.Supplier;
+
+import static java.util.Collections.unmodifiableList;
+
+public class Line {
+    private final List<StepStatus> stepStatuses;
+
+    public Line(Supplier<Boolean> generator, final int userCount) {
+        this.stepStatuses = createStepStatuses(generator, userCount);
+    }
+
+    private List<StepStatus> createStepStatuses(Supplier<Boolean> generator, final int userCount) {
+        final List<StepStatus> stepStatuses = new ArrayList<>(userCount - 1);
+        IntStream.range(0, userCount - 1)
+                .forEach(i -> stepStatuses.add(generateStepStatus(generator, stepStatuses, i)));
+        return stepStatuses;
+    }
+
+    private StepStatus generateStepStatus(
+            Supplier<Boolean> generator,
+            final List<StepStatus> stepStatuses,
+            final int index) {
+        if (index > 0 && stepStatuses.get(index - 1).isExist()) {
+            return StepStatus.getStepStatus(false);
+        }
+        return StepStatus.getStepStatus(generator.get());
```

Comment:
```suggestion
            return StepStatus.NONE;
        }
        return StepStatus.getStepStatus(generator.get());
```
앞에서도 이야기했지만 true, false를 사용하는것과 별반 차이가 없어보여요.

Reply:
`StepStatus.getStepStatus(false);` 이 부분은 `NONE`으로 변경해 주었습니다. 😊